### ISSUE TITLE
The "next" callback was never called inside of "AuthenticationGuard"

### ISSUE THREAD

**Issue by upwebdesign** [jnt0r/vue-auth0-plugin#102]
@jnt0r When using a guarded route (while not logged in) I have been running into an issue where `The "next" callback was never called inside of "AuthenticationGuard"`

Ive seen some solutions to this including creating a callback as an argument so next is always called.

```ts
const verify = async () => {
  next(() => {
    if (Plugin.state.authenticated) {
      return true;
    }
    return Plugin.properties.loginWithRedirect({ appState: { targetUrl: to.fullPath } });
  })
};
```

https://github.com/jnt0r/vue-auth0-plugin/blob/d91214e63d2900ebd0bb0f2050f9d678f6311bea/src/AuthenticationGuard.ts#L11-L16

Im not sure what the code suggestion will do, but thought I would have you take a look at it.

**Comment by jnt0r** [comment]
Thanks for mentioning that.
Just took a look into the documentation of vue-router and learned, that using next is optional in v4 because it leads to many mistakes: [docu](https://next.router.vuejs.org/guide/advanced/navigation-guards.html#optional-third-argument-next)

So I would rather like to get rid of the next parameter. I will take a look at this on the weekend.

**Comment by upwebdesign** [comment]
@jnt0r to add to this and I dont know if it is related, after this message, we are receiving `[Vue Router warn]: uncaught error during route navigation`, `Error: Invalid navigation guard` and `Unexpected error when starting the router: Error: Invalid navigation guard` errors when attempting to reload a page that requires authentication and we are already authenticated. Here is a screenshot:

![image](https://user-images.githubusercontent.com/1845130/126835356-aa15b84a-283d-435d-9cd7-f090349e878c.png)

**Comment by jnt0r** [comment]
Just released a fix for this in 2.1.5. Can you confirm that the problem is resolved?

**Comment by upwebdesign** [comment]
@jnt0r I just confirmed that this warning is no longer showing in my console. Thank you for the quick update to this!
